src/07-uart/send-a-single-byte.md

```diff
@@ -9,40 +9,56 @@ In order to do that we will use the following snippet (this one is already in `0
 {{#include src/main.rs}}
 ```
 
+The most prevalent new thing here is obviously the `cfg` directives to conditionally include/exclude
+parts of the code. This is mostly just because we want to work with a regular UART for the micro:bit v1
+and with the UARTE for micro:bit v2.
 
-There are some parts we have already seen before but also lots of new stuff so lets focus on those.
-First things first, how do we know that we have to interact with `P0_24` and `P0_25` here? If you take a look
-at the micro:bit schematics again, in page 5 you will notice that those two pins are also referred to as `TGT_RX`
-and `TGT_TX`. Not only that, they are also connected to our debug probe (see page 2) so those are most likely the two
-pins we are looking for.
-
-Next up, we can observe a new pin configuration we haven't seen before `p0.p0_25.into_floating_input()`. This just
-means that `P0_25` is now an input. What exactly floating means is none of our concern right now. But how do we know
-we have to put our pins in this exact mode? If you look at the line below you can see that we construct an instance
-of `Pins` which is later passed on to our UART peripheral constructor (`cts` and `rts` are for more advanced UART features
-we do not use here, hence they are set to `None`) and the [type signature](https://docs.rs/nrf51-hal/0.11.0/nrf51_hal/uart/struct.Pins.html)
-of `Pins` already tells us what to do.
-
-Next, we construct our UART peripheral with `hal::Uart::new(p.UART0, pins, hal::uart::Parity::EXCLUDED, hal::uart::Baudrate::BAUD115200);`.
-This function takes ownership of `p.UART0` and our `pins` so nobody else can mess with either the UART peripheral or our pins while
+You will also have noticed that this is the first time we are including some code that is not from a library,
+namely the `serial_setup` module. Its only purpose is to provide a nice and usable wrapper around the UARTE
```

Comment:
```suggestion
namely the `serial_setup` module. Its only purpose is to provide a nice wrapper around the UARTE
```
The implication that without this wrapper the UARTE would be somewhat unusable seems harsh.